src/js/components/devices/device-details/devicetwin.js

```diff
@@ -130,7 +131,13 @@ export const DeviceTwin = ({ device, getDeviceTwin, integrations, setDeviceTwin
   const [isSync, setIsSync] = useState(true);
   const editorRef = useRef(null);
 
-  const integration = integrations.find(integration => integration.provider === externalProvider.provider);
+  var integrationIds = Object.keys(device.twinsByIntegration ?? {}).filter(n => n != 'twinError');
+  if (integrationIds.length == 0) {
+    integrationIds = integrations.filter(integration => EXTERNAL_PROVIDER[integration.provider]?.deviceTwin).map(integration => integration.id);
+  }
```

Comment:
with both remaining providers offering device twin capabilities this should no longer be needed...

Reply:
We'll soon have Webhooks, which do NOT support device twin capabilities.

Reply:
ah ok, I did not expect Webhooks to be handled in that area...

Reply:
Web Hooks will be "yet another integration", but without support for the device state (twin/shadow).
We expect to configure webhooks using the same "integrations" API end-points.